src/lib.rs

```diff
@@ -267,6 +287,9 @@ where
         // Accumulate changes from requested to last recorded
         let mut full = Vec::new();
         for id in kv.changes_store.id_queue.iter().skip(id_position).rev() {
+            if id == &requested_id {
+                break;
+            }
```

Comment:
I think you can continue with iterator syntax here using `.take_while(|id| id != requested_id)`

Reply:
And the call `.for_each(|id| full.extend((...))` if we want to do 100% iterator syntax

Reply:
Done